R/slice-chop.R

```diff
@@ -89,8 +105,25 @@
 #'   ave2(breaks, wool, mean),
 #'   ave(breaks, wool, FUN = mean)
 #' )
-vec_chop <- function(x, indices = NULL) {
-  .Call(ffi_vec_chop, x, indices)
+#'
+#' # If you know your input is sorted and you'd like to split on the groups,
+#' # `vec_run_sizes()` can be efficiently combined with `sizes`
+#' df <- data_frame(
+#'   g = c(2, 5, 5, 6, 6, 6, 6, 8, 9, 9),
+#'   x = 1:10
+#' )
+#' vec_chop(df, sizes = vec_run_sizes(df$g))
+#'
+#' # If you have a list of homogeneous vectors, sometimes it can be useful to
+#' # unchop, apply a function to the flattened vector, and then rechop according
+#' # to the original indices. This can be done efficiently with `list_sizes()`.
+#' x <- list(c(1, 2, 1), c(3, 1), 5, double())
+#' x_flat <- list_unchop(x)
+#' x_flat <- x_flat + max(x_flat)
+#' vec_chop(x_flat, sizes = list_sizes(x))
+vec_chop <- function(x, indices = NULL, ..., sizes = NULL) {
+  check_dots_empty0(...)
```

Comment:
I don't think we can move `indices` at this point, but I wanted to add the empty dots in and felt it was worth it to do so anyways

Reply:
Good idea, done here https://github.com/r-lib/vctrs/pull/1813/commits/c1bd65e261efcba34b7813d5e4f38abd02287c85